ibridge_controller.php

```diff
@@ -8,62 +8,54 @@
  **/
 class Ibridge_controller extends Module_controller
 {
-
 	/*** Protect methods with auth! ****/
 	function __construct()
 	{
-		// Store module path
-		$this->module_path = dirname(__FILE__);
+        // Store module path
+        $this->module_path = dirname(__FILE__);
 	}
 
 	/**
 	 * Default method
 	 * @author tuxudo
 	 *
 	 **/
-	function index()
-	{
-		echo "You've loaded the ibridge module!";
-	}
-    
+    function index()
+    {
+        echo "You've loaded the ibridge module!";
+    }
+
      /**
-     * Get iBridge models for widget
+     * Get data for scroll widget
      *
      * @return void
      * @author tuxudo
      **/
-     public function get_ibridge()
-     {
-        $obj = new View();
+    public function get_scroll_widget($column)
+    {
+        $sql = "SELECT COUNT(CASE WHEN ".$column." <> '' AND ".$column." IS NOT NULL THEN 1 END) AS count, ".$column." 
+                FROM ibridge
+                LEFT JOIN reportdata USING (serial_number)
+                ".get_machine_group_filter()."
+                AND ".$column." <> '' AND ".$column." IS NOT NULL 
```

Comment:
Inserting $column in an $sql query like this is a SQL injection vulnerability. Check https://github.com/munkireport/softwareupdate/blob/master/softwareupdate_controller.php#L53 for a safer method. We might even put that in the base controller

Reply:
Fixed in latest PR with alphanumeric filtering